tools/aws_lab_setup/sample-vars-networking.yml

```diff
@@ -0,0 +1,17 @@
+ec2_region: us-east-1                 # region where the nodes will live
+ec2_az: us-east-1a                    # availability zone
+ec2_name_prefix: SEANTEST1120         # name prefix for all the VMs
+admin_password: ansible
+## Optional Variables
+#ec2_vpc_id: vpc-1234aaaa              # EC2 VPC ID in your region, if not supplied will be automatically created
```

Comment:
This is the next variable (vpc_subnet_id) should be required.

Reply:
what is the question here?  The vpc_subnet_id has a default values under roles/manage_ec2_instances/defaults and it is highly encouraged NOT to change this for the networking workshop because the instructions won't match.  I am fine with not even letting people change it.

Reply:
um. not following you here. The default value doesn't work for everyone -- only the people who have access to a specific account that are running in a specific region correct? Why would we default to something most cannot access? I'd prefer we say it is required and fail fast because that var isn't set.

Reply:
@tima I made an assumption here if the person that runs the provisioner does not specify an AWS VPC, that they also didn't have (or know what) a AWS VPC subnet id is.  So what I do is if the AWS VPC is not specified the provisioner will also create the subnet for you->

```
  ec2_vpc_subnet:
    region: "{{ ec2_region }}"
    az: "{{ ec2_az }}"
    vpc_id: "{{ create_vpc.vpc.id }}"
    cidr: "{{ ec2_subnet }}"
    resource_tags:
      Name: "{{ ec2_name_prefix }}-subnet"
  register: create_subnet
  when: ec2_vpc_id is undefined
```

this is located under: tools/aws_lab_setup/roles/manage_ec2_instances/tasks/provision.yml

so there is no default value for the subnet id, does that make sense? or am I answering the wrong question.

Reply:
Sort of. 

Let me back up a second. Despite what the provisioner in master is doing now (leftover from when this tool was for a few internal Ansible folks doing training in the US that had access to the same AWS account), the provisioner should not default to a specific VPC ID so that is required. (I think we have that addressed already now.) The user running the provisioner needs their own account and to (until until we add the ability to create one) setup their own VPC, subnet etc and feed the IDs of those resources into the provisioner. In this case these variables are required.

That aside, looking again at your PR, it looks like you've started creating those if they don't exist though you do that yourself rather than let Ansible do the work. (Sorry this comment pertains to another part of the content.) Ansible is a desired state engine so you just tell it "I need a VPC like this" and Ansible will figure out if that VPC exists and do it's thing. Same thing for the subnet, internet gateway etc.  I don't see why you need conditional logic like `when: ec2_vpc_id is undefined`. I sort of get `ec2_vpc_id2 is undefined` because of what is needed for the network automation labs though I think this could have been structured better -- that is neither here-not-there at the moment. Probably the `networking` conditional is sufficient on that second VPC for now.

So my feedback is 1) let Ansible's desired state capabilities figure out if it needs to create a VPC or related resource and 2) let Ansible lookup and register the IDs rather than have a user explicitly define those IDs. Even when a VPC exists it returns the ID it matched with.

Does that feedback make sense?

Reply:
>Does that feedback make sense?

No, because this part is not true->

>2) let Ansible lookup and register the IDs rather than have a user explicitly define those IDs

the user does not have to define anything... ansible does already take care of everything.  All you need to do is this->
```
ec2_region: us-east-1                 
ec2_az: us-east-1a                   
ec2_name_prefix: SEANTEST1120         
admin_password: ansible
email: no                              
student_total: 1                     
```

Honestly I am fine with deprecating the old method and it would make it much simpler.  For workshops it will just use a new VPC just for the workshop...

Reply:
I'm fine with deprecating. Honestly, I'm really tempted to say just rip it out, but that may be too reckless a thing.